jax/lib/xla_bridge.py

```diff
@@ -148,28 +151,38 @@ def register_backend_factory(name, factory, *, priority=0):
 
 
 if jax.lib._xla_extension_version >= 23:
-   register_backend_factory('interpreter', xla_client.make_interpreter_client,
-                            priority=-100)
-   register_backend_factory('cpu', xla_client.make_cpu_client,
-                            priority=0)
-   register_backend_factory('tpu_driver', _make_tpu_driver_client,
-                            priority=100)
-   register_backend_factory('gpu', xla_client.make_gpu_client,
-                            priority=200)
-   register_backend_factory('tpu', xla_client.make_tpu_client,
-                            priority=300)
+  register_backend_factory('interpreter', xla_client.make_interpreter_client,
+                           priority=-100)
+  if jax.lib._xla_extension_version >= 24:
+    if FLAGS.jax_cpu_backend_variant == 'stream_executor':
+      register_backend_factory('cpu', xla_client.make_cpu_client,
+                               priority=0)
+    else:
+      assert FLAGS.jax_cpu_backend_variant == 'tfrt'
+      register_backend_factory('cpu', xla_client.make_tfrt_cpu_client,
+                               priority=0)
+  else:
+    register_backend_factory('cpu', xla_client.make_cpu_client,
+                             priority=0)
+  register_backend_factory('cpu', xla_client.make_cpu_client,
```

Comment:
Should this registration still be here? I think it's subsumed by the ones above.

Reply:
removed